src/lib/components/BucketAggregation/BucketAggregationValues.js

```diff
@@ -67,16 +67,15 @@ class BucketAggregationValues extends Component {
     const valuesCmp = buckets.map((bucket) => {
       const isSelected = this._isSelected(
         this.aggName,
-        bucket.key,
+        bucket.key_as_string ? bucket.key_as_string : bucket.key,
```

Comment:
```suggestion
const key_field = bucket.key_as_string ? bucket.key_as_string : bucket.key
const isSelected = this._isSelected(this.aggName, bucket["key_field"], selectedFilters)
```

Reply:
I think you meant to write `bucket[key_field]` without quotes, right?